psi4/src/psi4/scfgrad/response.cc

```diff
@@ -549,35 +554,95 @@ std::shared_ptr<Matrix> RSCFDeriv::hessian_response() {
 
             for (int P = 0; P < basisset_->nshell(); P++) {
                 for (int Q = 0; Q < basisset_->nshell(); Q++) {
-                    int aP = basisset_->shell(P).ncenter();
-                    int aQ = basisset_->shell(Q).ncenter();
+                    const auto & shellP = basisset_->shell(P);
+                    const auto & shellQ = basisset_->shell(Q);
+                    int aP = shellP.ncenter();
+                    int aQ = shellQ.ncenter();
                     Vint->compute_shell_deriv1(P, Q);
-                    int nP = basisset_->shell(P).nfunction();
-                    int nQ = basisset_->shell(Q).nfunction();
-                    int oP = basisset_->shell(P).function_index();
-                    int oQ = basisset_->shell(Q).function_index();
-                    const double* buf_x = &buffer[3 * A * nP * nQ + 0 * nP * nQ];
-                    const double* buf_y = &buffer[3 * A * nP * nQ + 1 * nP * nQ];
-                    const double* buf_z = &buffer[3 * A * nP * nQ + 2 * nP * nQ];
-
+                    const auto &buffers = Vint->buffers();
+                    int nP = shellP.nfunction();
+                    int nQ = shellQ.nfunction();
+                    int oP = shellP.function_index();
+                    int oQ = shellQ.function_index();
+
+                    // buffer ordering is [Px, Py, Pz, Qx, Qy, Qz, A1x, A1y, A1z, A2x, ... ANy, ANz]
+                    // where the Px is the x derivative of shell P and A1x is the derivative w.r.t.
+                    // the nuclear charge located on atom1.  There are therefore 6 + 3*natoms buffers.
```

Comment:
A derivative with respect to _nuclear charge_? This function does CPHF, right? Why on earth do you need such a derivative? The CPHF equations just require nuclear position derivatives and orbital derivatives. Nuclear charge derivatives should never show up.

Reply:
The wording here is imprecise, it should state: "derivative wrt the coordinates of the nuclear charge".

Reply:
I'm still learning English, apparently :man_facepalming: